config/locales/models_and_forms.en.yml

```
@@ -60,13 +60,17 @@ en:
             actual_end_date:
               between: Date must be between %{min} years ago and %{max} years in the future
               not_in_future: Actual end date must not be in the future
+              zero: Date may not contain 0
```

```suggestion
              zero: Date component may not be 0
```

I would expect `07` to be valid.

---

Or even "{Day,Month,Year}" may not be 0"

---

Agree!